src/plugins/mission/mission_impl.cpp

```diff
@@ -1063,6 +1063,61 @@ void MissionImpl::pause_mission_async(const Mission::result_callback_t& callback
         std::bind(&MissionImpl::receive_command_result, this, std::placeholders::_1, callback));
 }
 
+void MissionImpl::clear_mission_async(const Mission::result_callback_t& callback)
+{
+    bool should_report_mission_result = false;
+    {
+        std::lock_guard<std::mutex> lock(_activity.mutex);
+        if (_activity.state == Activity::State::ABORTED) {
+            _activity.state = Activity::State::NONE;
+        }
+
+        if (_activity.state != Activity::State::NONE) {
+            should_report_mission_result = true;
+        }
+    }
+
+    if (should_report_mission_result) {
+        report_mission_result(callback, Mission::Result::BUSY);
+        return;
+    }
+
+    _parent->register_timeout_handler(
+        std::bind(&MissionImpl::process_timeout, this), RETRY_TIMEOUT_S, &_timeout_cookie);
+
+    {
+        std::lock_guard<std::mutex> lock(_activity.mutex);
+        _activity.state = Activity::State::MISSION_CLEAR;
+    }
+    {
+        std::lock_guard<std::recursive_mutex> lock(_mission_data.mutex);
+        _mission_data.result_callback = callback;
+        _mission_data.retries = 0;
+    }
+
+    clear_mission();
+}
+
+void MissionImpl::clear_mission()
+{
+    mavlink_message_t message;
+    mavlink_msg_mission_clear_all_pack(
+        _parent->get_own_system_id(),
+        _parent->get_own_component_id(),
+        &message,
+        _parent->get_system_id(),
+        _parent->get_autopilot_id(),
+        MAV_MISSION_TYPE_MISSION);
+
+    _parent->send_message(message);
```

Comment:
Found the mistake! Remove this line, otherwise you send the message twice!

Reply:
Good catch!